samza-azure/src/main/java/org/apache/samza/system/azureblob/BlobClientBuilder.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.system.azureblob;
+
+import com.azure.storage.blob.BlobServiceAsyncClient;
+
+
+/**
+ * Create a BlobServiceAsyncClient. Implementation controls construction of
```

Comment:
lets add info about what is required of the class implemting this interface -- like needs to be thread safe, will need to safely close the builder etc

Reply:
@lakshmi-manasa-g , I believe there are no additional guarantees required for the customer implementing this interface. for instance, Thread safety needn't be a constraint here. 
Mostly these changes are only refactoring work proposed from our end, & the actual underlying functionality code still remains the same. Let me know your thoughts are aligned. 

cc: @ehoner

Reply:
from the current PR changes perspective, its only code refactoring. but from the interface pov, there is a new interface that samza users can implement and need some guidelines on thread safety of the new impl they will add to their usage. 
for this particular interface, it looks like this --> system producer is a single instance per samza container and hence there will be a single instance of this client that will be instantiated per container and all the tasks (+ threads) within the container will use the same instance of the client. Hence the client impl NEEDS to be thread safe. 

for the purpose of this PR, there is no code change needed since its only a refactor and uses the pre-existing impl of the client. so please add the above thread safety requirements along with info on how the client is closed in the system producer and if a new impl of this interface needs to do something special abt closing the client

Reply:
Sure @lakshmi-manasa-g , catered the above suggestion. Can you kindly check once.